engine/src/main/java/io/zeebe/engine/processing/streamprocessor/ProcessingStateMachine.java

```diff
@@ -167,7 +167,7 @@ public ProcessingStateMachine(
 
     final int partitionId = logStream.getPartitionId();
     typedEvent = new TypedEventImpl(partitionId);
-    responseWriter = new TypedResponseWriterImpl(context.getCommandResponseWriter(), partitionId);
+    responseWriter = new TypedResponseWriterImpl(context.getWriters().response(), partitionId);
```

Comment:
I am wondering why the writers object doesn't contain a `TypedResponseWriter` directly. I find it somewhat cumbersome that this object has to be wrapped again and we have to find out the `partitionId` from somewhere else.

This is currently the only place which calls the method. So I think it could be changed. Also the `partitionId` comes from  `context.getLogStream().getPartitionId()`, so it is available in `context.getWriters()` as far as I can see.

What do you think?

Reply:
>the writers object doesn't contain a TypedResponseWriter directly

I understand, but the same could be said for the use of the `TypedStreamWriter logStreamWriter` in the same class. Let's add a note to the clean up issue to have another look at it there. For now, all writers are available to the processors we want to migrate.

>so it is available in context.getWriters()

I feel like the partition id should not be something related to the writers. The `LogStream` has a partition id, but the writers don't need to know anything about a partition, just where to write.

Reply:
Commented here: https://github.com/zeebe-io/zeebe/issues/6202#issuecomment-777362202